SignalShareExtension/ShareViewController.swift

```diff
@@ -364,19 +364,21 @@ public class ShareViewController: UINavigationController, ShareViewDelegate, SAE
     // MARK: Helpers
 
     private func presentConversationPicker() {
-        // pause any animation revealing the "loading" screen
-        self.view.layer.removeAllAnimations()
-
-        // Once we've presented the conversation picker, we hide the loading VC
-        // so that it's not revealed when we eventually dismiss the share extension.
-        loadViewController.view.isHidden = true
-
         self.buildAttachment().then { attachment -> Void in
             let conversationPicker = SharingThreadPickerViewController(shareViewDelegate: self)
             let navigationController = UINavigationController(rootViewController: conversationPicker)
             navigationController.isNavigationBarHidden = true
             conversationPicker.attachment = attachment
-            self.present(navigationController, animated: true, completion: nil)
+            if let presentedViewController = self.presentedViewController {
+                Logger.debug("\(self.logTag) dismissing \(presentedViewController) before presenting conversation picker")
+                self.dismiss(animated: true) {
+                    self.present(navigationController, animated: true)
```

Comment:
If the loading modal is presented, we dismiss it, and then immediately present the conversation picker.

This isn't a great look. Any suggestions?

Reply:
`ShareViewController` is a `UINavigationController`.  Why are we creating a new `UINavigationController`?  I would just do:
```
        self.setViewControllers([conversationPicker], animated: false)
```
or `animated` could be true, but IMO our users would rather skip the animation.  People are in a hurry when they're trying to share.